File: Paris/2015/2015-02-19-Chatroom-Erlang-Randori/hiker.erl
-module(hiker).
-export([chatroom/0,enter/3]).
chatroomLoop(ClientPids) ->
  receive
    {enter, Name, Pid} -> NewClientPids = [Pid|ClientPids],
                          broadcast_message(NewClientPids , Name ++ " entered the room"),
                          chatroomLoop(NewClientPids)
    _ -> chatroomLoop(ClientPids)
  end.
broadcast_message(ClientPids, Message) ->
  lists:foreach(
    fun(ClientPid) -> ClientPid! {message, Message} end,
    ClientPids
  ).
 
chatroom() ->
  spawn(fun() -> chatroomLoop([]) end).
enter(Room, Name) ->
  Room! {enter, Name, self()}.

File: Paris/2015/2015-02-12 - Chatroom - Go - Randori/hiker.go
package chatroom
type Chatroom struct {
    clients []Client
}
func newChatroom() *Chatroom{
    return new(Chatroom)
}
func (room *Chatroom) enter(newClient Client) {
    room.clients = append(room.clients,newClient)
    for _,client := range room.clients {
    client.notify(newClient.name() + " entered the room")
    }  
}
type Client interface {
    notify(message string)
    name() string
}